src/main/java/com/stosik/parking/reservation/domain/ParkingStore.java

```diff
@@ -0,0 +1,41 @@
+package com.stosik.parking.reservation.domain;
+
+import com.stosik.parking.reservation.domain.model.Reservation;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+@Repository
+ class ParkingStore
+{
+    private ConcurrentHashMap<String, Reservation> store = new ConcurrentHashMap<>();
+    
+    public Reservation save(Reservation reservation)
+    {
+        store.put(reservation.getCarLicenseId(), reservation);
+        return reservation;
+    }
+    
+    public Optional<Reservation> findById(String title)
+    {
+        return Optional.of(store.get(title));
+    }
+    
+    public Reservation delete(Reservation reservation)
+    {
+        return store.remove(reservation.getCarLicenseId());
+    }
+    
+    public List<Reservation> findAll()
```

Comment:
jaki sens biznesowy ma zwracanie wszystkich obiektów z DB? starczy pamięci w normalnej aplikacji?

Reply:
Parking store nie operuje na bazie a obiektach (rezerwacjach) które się jeszcze nie skończyły i są trzymane w pamięci (służy on jako event store)

Reply:
* Co, gdy aplikacja nagle się ubije, albo będzie potrzebny restart?
* Co to znaczy, że służy jako event store?

Reply:
Generalnie jak aplikacja stanie to wszystkie rezerwacje ktore sie nie zakonczyly w tym momencie znikna, chcialem tutaj zaprezentowac bardziej pewien koncept ktory faktycznie powinienem lepiej przemyslec, wydaje mi sie po prostu ze persystowanie do bazy danych takich danych wiaze sie z duza iloscia zapytan i mozna by w inny sposob zooptymalizowac ciagle I/O. 
Przez Event Store mam na mysli miejsce w ktorych przechowuje zdarzenia jakie zaszly w systemie, wyobrazilem sobie ze taka rezerwacja to moze byc wlasnie takie zdarzenie ktore warto odnotowywac ale nie koniecznie pchac od razu do bazy w momencie rozpoczecia.